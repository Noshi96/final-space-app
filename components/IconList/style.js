import styled from 'styled-components'
import Icon from '../Icon/Icon'

export const StyledIcon = styled(Icon)`
  margin-right: 0.8rem;
  color: ${(props) =>
    props.status === 'Alive'
      ? '#D32F2F;'
      : props.status === 'Deceased'
      ? '#2196F3;'
      : '#000;'};
`

export const List = styled.ul`
  grid-row: 4 / 5;
  display: flex;
  flex-direction: column;
  gap: 0.85rem;
  margin-top: 1.5rem;
  margin-right: 1rem;
  padding-right: 2rem;
  padding-left: 2rem;
  text-decoration: none;
`

export const IconListElement = styled.li`
  width: 100%;
  display: flex;
  align-items: center;
`
