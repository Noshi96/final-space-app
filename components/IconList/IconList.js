import { List, IconListElement, StyledIcon } from './style'

const IconList = ({ origin, status, species, occurrence }) => {
  return (
    <List>
      <IconListElement>
        <StyledIcon name="public" />
        {origin !== 'Unknown' ? `Born on ${origin}` : 'Unknown'}
      </IconListElement>
      <IconListElement>
        {status === 'Alive' ? (
          <StyledIcon status={status} name="favorite" />
        ) : status === 'Deceased' ? (
          <StyledIcon status={status} name="heart_broken" />
        ) : (
          <StyledIcon name="question_mark" />
        )}
        Is {status.toLowerCase()}
      </IconListElement>
      <IconListElement>
        <StyledIcon name="group" />
        {species ?? 'Unknown'}
      </IconListElement>
      <IconListElement>
        <StyledIcon name="video_file" />
        Was seen in {occurrence} chapters
      </IconListElement>
    </List>
  )
}

export default IconList
