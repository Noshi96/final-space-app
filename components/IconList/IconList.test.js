/* globals describe, expect, it */

import { render } from '@testing-library/react'
import IconList from './IconList'

describe('Icon', () => {
  const singleCharacter = {
    id: 1,
    name: 'Gary Goodspeed',
    status: 'Alive',
    species: 'Human',
    gender: 'Male',
    hair: 'Blonde',
    origin: 'Earth',
    abilities: [
      'Piloting',
      'Marksmanship',
      'Hand-to-hand combat',
      'Weapons: Blasters',
      'Weapons: Black hole gun (briefly)',
      'Weapons: Robot arm',
      'Weapons: Laser gun',
      'Weapons: Laser sword',
    ],
    img_url:
      'https://finalspaceapi.com/api/character/avatar/gary_goodspeed.png',
  }
  const countOccurrence = 22

  it('renders IconList', () => {
    const iconList = render(
      <IconList
        origin={singleCharacter.origin}
        status={singleCharacter.status}
        species={singleCharacter.species}
        occurrence={countOccurrence}
      />
    )
    expect(iconList.container).toMatchSnapshot()
  })
})
