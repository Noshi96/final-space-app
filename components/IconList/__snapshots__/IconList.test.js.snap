// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders IconList 1`] = `
<div>
  <ul
    class="sc-c4342ac9-1 guIfcn"
  >
    <li
      class="sc-c4342ac9-2 kBpwcf"
    >
      <span
        class="material-icons sc-c4342ac9-0 jEsdHR"
      >
        public
      </span>
      Born on Earth
    </li>
    <li
      class="sc-c4342ac9-2 kBpwcf"
    >
      <span
        class="material-icons sc-c4342ac9-0 hGdQyi"
      >
        favorite
      </span>
      Is 
      alive
    </li>
    <li
      class="sc-c4342ac9-2 kBpwcf"
    >
      <span
        class="material-icons sc-c4342ac9-0 jEsdHR"
      >
        group
      </span>
      Human
    </li>
    <li
      class="sc-c4342ac9-2 kBpwcf"
    >
      <span
        class="material-icons sc-c4342ac9-0 jEsdHR"
      >
        video_file
      </span>
      Was seen in 
      22
       chapters
    </li>
  </ul>
</div>
`;
